src/vsr/client.zig

```diff
@@ -293,11 +266,58 @@ pub fn Client(comptime StateMachine_: type, comptime MessageBus: type) type {
 
         /// Releases a message back to the message bus.
         pub fn unref(self: *Self, message: *Message) void {
+            // Only request messages are unreferenced by this function,
+            // all others should use `message_bus.unref` directly.
+            assert(message.header.command == .request);
```

Comment:
This assumption that `unref` can only accept `.request` heavily depends on how `tb_client` works.
One using `vsr.client` directly can call `get_message` and then `unref` without setting the message header, causing a panic for accessing undefined memory.

Even so, I believe this assertion is worth keeping to avoid misuse between `client.unref`, `bus.unref` and `pool.unref`.

Reply:
Assertion removed, since `client.release` can be called by `errdefer` with an uninitialized message.